app/upgrades/v4.0.0/upgrades.go

```diff
@@ -57,3 +66,24 @@ func setMarketMapParams(ctx sdk.Context, marketmapKeeper *marketmapkeeper.Keeper
 	}
 	return marketmapKeeper.SetParams(ctx, marketmapParams)
 }
+
+func enableVoteExtensions(ctx sdk.Context, consensusKeeper *consensuskeeper.Keeper) error {
+	oldParams, err := consensusKeeper.Params(ctx, &types.QueryParamsRequest{})
+	if err != nil {
+		return err
+	}
+
+	// we need to enable VoteExtensions for Slinky
+	oldParams.Params.Abci = &comettypes.ABCIParams{VoteExtensionsEnableHeight: ctx.BlockHeight()}
```

Comment:
```suggestion
	oldParams.Params.Abci = &comettypes.ABCIParams{VoteExtensionsEnableHeight: ctx.BlockHeight() + 4}
```

we would prefer that the VE enabled height is set to some offset after the upgrade.

This will make it easier for all to debug the upgrade in case there are issues.  Only the state will be set here, all of that can be validated for a few blocks, and then the VEs can kick in.